atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -364,6 +365,50 @@ public DBType getDbType() {
         return connConfig.getDbType();
     }
 
+    @Override
+    public void setPassword(String newPassword) {
+        Preconditions.checkNotNull(newPassword, "password cannot be null");
+        State localState = state;
+        if (localState.type == StateType.ZERO) {
+            synchronized (this) {
+                // need to check again in case init just happened
+                localState = state;
+                if (localState.type == StateType.ZERO) {
+                    // the pool is not initialized, need to set password on the config object
+                    internalSetPassword(newPassword, hikariConfig);
+                    return;
+                }
+            }
+        }
+        if (localState.type == StateType.NORMAL) {
+            // pool is initialized, need to set password directly on the pool
+            internalSetPassword(newPassword, localState.dataSourcePool);
+        } else if (localState.type == StateType.CLOSED) {
+            throw new SafeRuntimeException("Hikari pool is closed", localState.closeTrace);
+        }
+    }
+
+    /**
+     * Sets the password (and username) on the HikariConfig (or HikariDataSource pool) if it is necessary.
+     *
+     * <p>Note that when the password matches static config, performance is better when setting the username and
+     * password to null so that hikari uses the original JDBC parameters and avoids making new copies for every
+     * new connection. See {@link com.zaxxer.hikari.pool.PoolBase#newConnection}. Also see
+     * {@link com.palantir.nexus.db.pool.config.OracleConnectionConfig#getHikariProperties} for the JDBC properties.
+     *
+     * <p>Hikari checks the username field for being null to determine if it needs to copy the JDBC parameters, so
+     * the username *must* be set for it to respect the password change.
+     */
+    private void internalSetPassword(String newPassword, HikariConfig configOrPool) {
```

Comment:
As above, I'm a little concerned about concurrency (but I might be being paranoid here).